gulp/tasks/gen-service-worker.js

```diff
@@ -86,13 +85,29 @@ gulp.task('gen-service-worker', () => {
         rootDir + '/fonts/roboto/Roboto-Bold.ttf',
         rootDir + '/images/card_media_player_bg.png',
       ],
-      runtimeCaching: [{
-        urlPattern: /\/static\/translations\//,
+      // Rules are proceeded in order and negative per-domain rules are not supported.
+      runtimeCaching: [
+        { // Cache translations on first access
+          urlPattern: '/static/translations/*',
         handler: 'cacheFirst',
-      }, {
-        urlPattern: RegExp('^[^/]*/' + fallbackList + '.'),
+        },
+        { // Get static, api, and local (and home-assistant-polymer in dev mode) from network.
+          // Use cache only if Home Assistant server is unreachable.
+          urlPattern: '/(home-assistant-polymer|static|api|local)/*',
+          handler: 'networkOnly',
+        },
+        { // Get manifest and service worker from network.
+          // Use cache only if Home Assistant server is unreachable.
+          urlPattern: '/(service_worker.js|manifest.json)',
```

Comment:
I don't think that a service worker can cache the service worker? That would mean that you can get into a state where one can never upgrade ;-)

Reply:
I don't know, but it can't hurt